src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -357,6 +364,8 @@ static void verifyColumnNames(Connection conn, String sqlConfig, String sql,
               "Matched column \"{0}\" as \"{1}\" in query {2}",
               new Object[] { match, actual, sqlConfig });
           targets.remove(match);
+          // TODO(bmj): What happens with MySQL here.
```

Comment:
This TODO can go away. MySQL doesn't have a problem here. If some other database does, then either we can decide we want a separate try-catch here, to allow the unique key types to fail independently of the rest of the validation (which is already explicitly not failing by time we get here), or just ignore errors here and bail on the validation.